Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -3458,50 +3459,185 @@ def panorama_delete_url_filter_command(url_filter_name: str):
+
+    value = xml_dict.get(key)
+
+    if type(value) is dict:
+        return next(
+            (xml_get(value, sub, subkeys)
+             for sub in subkeys
+             if sub in value),
+            None
+        ) or next(
+            (xml_get(value, k)
+             for k in value.keys()
+             if not str(k).startswith('@')),
+            None
+        )
+    if type(value) is list:
+        return [
+            xml_get({0: item}, 0, subkeys)
+            for item in value
+        ]
+
+    return value
+
+
 def prettify_rule(rule: dict):
-    pretty_rule = {
-        'Name': rule['@name'],
-        'Action': rule['action']
-    }
-    if DEVICE_GROUP:
-        pretty_rule['DeviceGroup'] = DEVICE_GROUP
-    if '@loc' in rule:
-        pretty_rule['Location'] = rule['@loc']
-    if isinstance(rule.get('category'), dict) and 'member' in rule['category']:
-        pretty_rule['CustomUrlCategory'] = rule['category']['member']
-    if isinstance(rule.get('application'), dict) and 'member' in rule['application']:
-        pretty_rule['Application'] = rule['application']['member']
-    if isinstance(rule.get('destination'), dict) and 'member' in rule['destination']:
-        pretty_rule['Destination'] = rule['destination']['member']
-    if isinstance(rule.get('from'), dict) and 'member' in rule['from']:
-        pretty_rule['From'] = rule['from']['member']
-    if isinstance(rule.get('service'), dict) and 'member' in rule['service']:
-        pretty_rule['Service'] = rule['service']['member']
-    if isinstance(rule.get('to'), dict) and 'member' in rule['to']:
-        pretty_rule['To'] = rule['to']['member']
-    if isinstance(rule.get('source'), dict) and 'member' in rule['source']:
-        pretty_rule['Source'] = rule['source']['member']
-    if isinstance(rule.get('tag'), dict) and 'member' in rule['tag']:
-        pretty_rule['Tags'] = rule['tag']['member']
-    if isinstance(rule.get('log-setting'), dict) and '#text' in rule['log-setting']:
-        pretty_rule['LogForwardingProfile'] = rule['log-setting']['#text']
-    if disabled := rule.get('disabled'):
-        pretty_rule['Disabled'] = disabled
 
-    return pretty_rule
+    rule_get = partial(xml_get, rule)
+
+    tick_box: Dict[Any, str] = {
+        'yes': '✅',
+        'no': '⬜',
+        None: '⬜',
+    }
+
+    context_rule: Dict[str, Any] = {
+        'NegateDestination': rule_get('negate-destination'),
+        'Disabled': rule_get('disabled'),
+        'ICMPUnreachable': rule_get('icmp-unreachable'),
+        'Description': rule_get('description'),
+        'ProfileSetting': rule_get('profile-setting'),
+        'GroupTag': rule_get('group-tag'),
+        'Schedule': rule_get('schedule'),
+        'LogStart': rule_get('log-start'),
+        'LogForwardingProfile': rule_get('log-setting'),
+        'NegateSource': rule_get('negate-source'),
+    }
+    pretty_rule: Dict[str, Any] = {}
+
+    context_rule['DeviceGroup'] = DEVICE_GROUP or None
```

Comment:
```suggestion
    context_rule['DeviceGroup'] = DEVICE_GROUP
```

`None` not needed, its empty string by default